src/backend/parser/gram.y

```diff
@@ -15551,10 +15551,16 @@ func_expr_common_subexpr:
 				}
 			| TRIM '(' trim_list ')'
 				{
-					$$ = (Node *) makeFuncCall(SystemFuncName("btrim"),
-											   $3,
-											   COERCE_SQL_SYNTAX,
-											   @1);
+					if (sql_dialect == SQL_DIALECT_TSQL)
+						$$ = (Node *) makeFuncCall(list_make2(makeString("sys"), makeString("trim")),
+													$3,
```

Comment:
Do we need to really introduce engine parser change, can we explore some non-invasive options

Reply:
1. +1 to above, can't we identify TRIM node in ANTLR and modify it to sys.TRIM even before it comes here.
2. Also, what about LEADING/TRAILING/BOTH variants of `TRIM`? don't we need to handle them as well?

Reply:
1. removed engine parser change
2. we currently do not support `LEADING/TRAILING/BOTH` in babelfish, this support will be added later separately.